library/system.lisp

```diff
@@ -61,12 +63,15 @@ While the result will always contain microseconds, some implementations may retu
                (cl:* 1000000 (cl:- end start))
                cl:internal-time-units-per-second)))))
 
-  (declare sleep (Integer -> Unit))
+  (declare sleep ((math:Rational :a) => :a -> Unit))
```

Comment:
This will probably cons more than people care for, and so we might add some specializations for floats and the like. Maybe file an issue for that? It would be a good first issue for somebody.

Reply:
#1307